centreon/packages/ui/src/Graph/BarStack/ResponsiveBarStack.tsx

```diff
@@ -0,0 +1,255 @@
+import { BarStack as BarStackVertical, BarStackHorizontal } from '@visx/shape';
+import { Group } from '@visx/group';
+import { useTooltip, useTooltipInPortal, defaultStyles } from '@visx/tooltip';
+import { localPoint } from '@visx/event';
+import numeral from 'numeral';
+import { Text } from '@visx/text';
+import { equals } from 'ramda';
+
+import { useTheme } from '@mui/system';
+
+import { LegendProps } from '../Legend/models';
+import { Legend as LegendComponent } from '../Legend';
+import { getValueByUnit } from '../common/utils';
+
+import { BarStackProps } from './models';
+import { useBarStackStyles } from './BarStack.styles';
+import useResponsiveBarStack from './useResponsiveBarStack';
+
+const DefaultLengd = ({ scale, configuration }: LegendProps): JSX.Element => (
+  <LegendComponent configuration={configuration} scale={scale} />
+);
+
+const BarStack = ({
+  title,
+  data,
+  width: barWidth,
+  height: barHeight,
+  onSingleBarClick,
+  displayLegend = true,
+  Tooltip,
+  legendConfiguration = { direction: 'column' },
+  Legend = DefaultLengd,
+  unit = 'Number',
+  displayValues,
+  variant = 'Vertical'
+}: BarStackProps & { height: number; width: number }): JSX.Element => {
+  const theme = useTheme();
+  const { classes } = useBarStackStyles();
+
+  const tooltipStyles = {
+    ...defaultStyles,
+    backgroundColor: theme.palette.background.tooltip,
+    minWidth: 60
+  };
+
+  const {
+    tooltipOpen,
+    tooltipLeft,
+    tooltipTop,
+    tooltipData,
+    hideTooltip,
+    showTooltip
+  } = useTooltip();
+
+  const { containerRef, TooltipInPortal } = useTooltipInPortal({
+    scroll: true
+  });
+
+  const {
+    height,
+    width,
+    colorScale,
+    input,
+    keys,
+    legendScale,
+    total,
+    xScale,
+    yScale
+  } = useResponsiveBarStack({
+    barHeight,
+    barWidth,
+    data,
+    unit,
+    variant
+  });
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.svgWrapper}>
+        {title && (
+          <div className={classes.title}>
+            {`${numeral(total).format('0a').toUpperCase()} `} {title}
+          </div>
+        )}
+        <div
+          className={classes.svgContainer}
+          style={{
+            height: height + 15,
+            width: width + 15
```

Comment:
```suggestion
            height: height + 16,
            width: width + 16
```
Stay on the MUI 8x8 grid